server/automute_preferences.go

```diff
@@ -0,0 +1,59 @@
+package main
+
+import (
+	"github.com/mattermost/mattermost/server/public/model"
+	"github.com/mattermost/mattermost/server/public/plugin"
+)
+
+func (p *Plugin) PreferencesHaveChanged(c *plugin.Context, preferences []model.Preference) {
+	p.updateAutomutingOnPreferencesChanged(c, preferences)
+}
```

Comment:
Interesting, it looks like this can be called both for a single user's and multiple users' preferences?

Reply:
It's only called with a single user's preferences at the moment, but unfortunately, I wrote the signature in a way that it could be called with multiple users' preferences on the server side, so I decided to continue writing it this way. I thought that it might be possible to call that endpoint with multiple users' preferences at some point for the case where the customer wanted to set this for multiple users at once, but it turns out that wasn't the case.